include/seqan3/range/container/small_string.hpp

```diff
@@ -334,6 +335,39 @@ class small_string : public small_vector<char, capacity_ + 1>
         return str();
     }
     //!\}
+
+    /*!\name Input/output
+     * \{
+     */
+
+    /*!\brief Formatted output for the seqan3::small_string.
+     * \param[in,out] os  The std::basic_ostream to write to.
+     * \param[in]     str The seqan3::small_string to read from.
+     * \returns `os`.
+     */
+    friend std::ostream & operator<<(std::ostream & os, small_string const & str)
```

Comment:
why is it not enough to be printable with our debug_stream?

Reply:
Because it mimics all the interfaces of std::string except input/output. I think this is not super important but simply more complete and thus less confusing.